src/components/Select.jsx

```diff
@@ -325,7 +311,8 @@ export default class Select extends Component {
 
   componentWillReceiveProps(properties) {
     if (properties.children) {
-      this.children = flatten(properties.children);
+      this.children = flattenReactChildren(properties.children);
+      this.options = filter(this.children, isOption);
     }
 
```

Comment:
this.children will always be an array, I could have kept is same is children.props, but I think its perfectly fine to keep it array, coz we can then have more optimized helpers.